src/main/java/io/javalin/PathBuilder.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Javalin - https://javalin.io
+ * Copyright 2017 David Åse
+ * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
+ */
+
+package io.javalin
+
+import java.net.URLEncoder
+
+class PathBuilder(private val baseUrl: String) {
+
+    private val queryParams = mutableMapOf<String, Array<out String>>()
+    private val pathParams = mutableMapOf<String, String>()
+
+    fun pathParam(key: String, value: String): PathBuilder {
+        if (!baseUrl.contains(key)) {
+            throw IllegalArgumentException("No path-param matching '$key'")
+        }
+        if (!key.startsWith(":")) {
```

Comment:
I am not sure that semicolon should be required here. It makes sense inin the url definition in routes, but in the code we can use just a key without ":".

Reply:
I thought it made the code a bit more readable. I'll think about it.